examples/user_guide/Performance_and_Debugging.ipynb

```diff
@@ -46,6 +46,71 @@
+    "\n",
+    "When to use which approach cannot be answered easily however as a general guide use `asyncio` when you are performing many slow IO bound operations simultaneously and use threading when you have a limited amount of connections or file operations. In some situations threading can also be useful for CPU intensive operations where the operation being executed [releases the GIL](https://realpython.com/async-io-python/), this includes many NumPy, Pandas and Numba functions.\n",
+    "\n",
+    "### Threading\n",
+    "\n",
+    "Threading in Panel can be enabled easily by setting the `config.nthreads` parameter. This will start a `ThreadPoolExecutor` with the specified number of threads (or if set to zero it will set the number of threads based on your system, i.e. `min(32, os.cpu_count() + 4)`). Whenever an event is generated or a periodic callback fires Panel will then automatically dispatch the event to the executor. If launching an application with `panel serve` you can configure this option with the `--num-threads` option.\n",
+    "\n",
+    "### AsyncIO\n",
+    "\n",
+    "When using Python>=3.8 you can use async callbacks wherever you would ordinarily use a regular synchronous function, e.g. you can use `pn.bind` on an async function:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import aiohttp\n",
+    "\n",
+    "widget = pn.widgets.IntSlider(start=0, end=10)\n",
+    "\n",
+    "async def get_img(index):\n",
+    "    async with aiohttp.ClientSession() as session:\n",
+    "        async with session.get(f\"https://picsum.photos/800/300?image={index}\") as resp:\n",
+    "            return pn.pane.JPG(await resp.read())\n",
+    "            \n",
+    "pn.Column(widget, pn.bind(get_img, widget))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Similarly you can attach asynchronous callbacks using `.param.watch`:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "widget = pn.widgets.IntSlider(start=0, end=10)\n",
+    "\n",
+    "image = pn.pane.JPG()\n",
+    "\n",
+    "async def update_img(event):\n",
+    "    async with aiohttp.ClientSession() as session:\n",
+    "        async with session.get(f\"https://picsum.photos/800/300?image={event.new}\") as resp:\n",
+    "            image.object = await resp.read()\n",
+    "            \n",
+    "widget.param.watch(update_img, 'value')\n",
+    "widget.param.trigger('value')\n",
+    "            \n",
+    "pn.Column(widget, image)"
+   ]
+  },
```

Comment:
Again, explicitly stating what the sequence of triggers and events would be here will help people understand how it works.